gprofiler/merge.py

```diff
@@ -92,33 +176,39 @@ def parse_perf_script(script: str):
             match = SAMPLE_REGEX.match(sample)
             if match is None:
                 raise Exception("Failed to match sample")
-            yield match.groupdict()
+            sample_dict = match.groupdict()
+
+            pid = int(sample_dict["pid"])
+            process_name = sample_dict["comm"]
+            stack = sample_dict["stack"]
+            if stack is not None:
+                pid_to_collapsed_stacks_counters[pid][collapse_stack(stack, process_name)] += 1
+            pid_to_name.setdefault(pid, process_name)
```

Comment:
Why `setdefault`?

Reply:
?

Reply:
To avoid checking if the PID already exists and setting it if it doesn't. It does all of that in one atomic operation.